src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -810,4 +815,14 @@ public String getBuildPlanAccessSecret() {
     public void generateAndSetBuildPlanAccessSecret() {
         buildPlanAccessSecret = UUID.randomUUID().toString();
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnectRelatedEntities() {
+        Stream.of(exerciseHints, testCases, staticCodeAnalysisCategories).filter(Objects::nonNull).forEach(Collection::clear);
```

Comment:
maybe (!) in the future this could be done by reflection. I'm not sure whether we want or do not want this in Artemis; but finding all collection fields and clearing them if they are not null is easy to implement and would ensure that no new collections will be missed in the future.